hooks/tk-multi-publish2/maya.basic/collector.py

```diff
@@ -0,0 +1,244 @@
+                        "label": "Set Project",
+                        "tooltip": "Set the Maya project",
+                        "callback": lambda: mel.eval('setProject ""')
+                    }
+                }
+            )
+
+    def collect_current_maya_session(self, parent_item):
+        """
+        Creates an item that represents the current maya session.
+
+        :param parent_item: Parent Item instance
+        :returns: Item of type maya.session
+        """
+
+        publisher = self.parent
+
+        # get the path to the current file
+        path = cmds.file(query=True, sn=True)
+
+        # determine the display name for the item
+        if path:
+            file_info = publisher.util.get_file_path_components(path)
+            display_name = file_info["filename"]
+        else:
+            display_name = "Current Maya Session"
+
+        # create the session item for the publish hierarchy
+        session_item = parent_item.create_item(
+            "maya.session",
+            "Maya Session",
+            display_name
+        )
+
+        # get the icon path to display for this item
+        icon_path = os.path.join(
+            self.disk_location,
+            os.pardir,
+            "icons",
+            "maya.png"
+        )
+        session_item.set_icon_from_path(icon_path)
+
+        # discover the project root which helps in discovery of other
+        # publishable items
+        project_root = cmds.workspace(q=True, rootDirectory=True)
+        session_item.properties["project_root"] = project_root
+
+        self.logger.info("Collected current Maya scene")
+
+        return session_item
+
+    def collect_alembic_caches(self, parent_item, project_root):
+        """
+        Creates items for alembic caches
+
+        Looks for a 'project_root' property on the parent item, and if such
+        exists, look for alembic caches in a 'cache/alembic' subfolder.
+
+        :param parent_item: Parent Item instance
+        :param str project_root: The maya project root to search for alembics
+        """
+
+        # ensure the alembic cache dir exists
+        cache_dir = os.path.join(project_root, "cache", "alembic")
+        if not os.path.exists(cache_dir):
+            return
+
+        self.logger.info(
+            "Processing alembic cache folder: %s" % (cache_dir,),
+            extra={
+                "action_show_folder": {
+                    "path": cache_dir
+                }
+            }
+        )
+
+        # look for alembic files in the cache folder
+        for filename in os.listdir(cache_dir):
+            cache_path = os.path.join(cache_dir, filename)
+
+            # do some early pre-processing to ensure the file is of the right
+            # type. use the base class item info method to see what the item
+            # type would be.
+            item_info = self._get_item_info(filename)
+            if item_info["item_type"] != "file.alembic":
```

Comment:
Yes, we did talk about it. I've created internal ticket #44090 to revisit. I'm not exactly sure what the API should look like, but at least now we can track/prioritize revisiting.